pkg/controller/clone/host-clone.go

```diff
@@ -195,6 +195,31 @@ func (p *HostClonePhase) createClaim(ctx context.Context) (*corev1.PersistentVol
 	}
 	cc.AddLabel(claim, cc.LabelExcludeFromVeleroBackup, "true")
 
+	if myVolumeMode := cc.GetVolumeMode(claim); myVolumeMode == corev1.PersistentVolumeFilesystem {
+		// It is possible when the source pvc has VolumMode 'block'
+		// and the claim has 'filesystem' in which case the filesystem overhead need to be considered
+		sourcePvc := &corev1.PersistentVolumeClaim{}
+		sourcePvcKey := client.ObjectKey{Namespace: p.Namespace, Name: p.SourceName}
+
+		if err := p.Client.Get(ctx, sourcePvcKey, sourcePvc); err != nil {
+			return nil, err
+		}
+
+		realSourcePvcSizeRequest := sourcePvc.Spec.Resources.Requests[corev1.ResourceStorage]
```

Comment:
This is the only thing I have some concern about, this works fine if the source volume mode is block, since we can just inflate the size of the source PVC to account for the overhead. But if the source volume mode is filesystem, it is likely the source is already inflated, and we don't want to inflate it again. I think in that case we should check if there is a datavolume associated with the PVC, and get the size from there.

Reply:
@awels  I've made some changes to improve it. Pls take a look. Thanks!